spec/jobs/jobs_base_spec.rb

```diff
@@ -47,4 +47,43 @@ def execute(args)
     ::Jobs::Base.new.perform('hello' => 'world', 'sync_exec' => true)
   end
 
+  context "with fake jobs" do
+    let(:common_state) { [] }
+
+    before do
+      class Jobs::TestJob1 < Jobs::Base
+        def execute(args)
+          @@state << "job_1_executed"
+        end
+      end
+      Jobs::TestJob1.class_variable_set(:@@state, common_state)
+
+      class Jobs::TestJob2 < Jobs::Base
+        def execute(args)
+          @@state << "job_2_started"
+          Jobs.enqueue(:test_job_1)
+          @@state << "job_2_finished"
+        end
+      end
+      Jobs::TestJob2.class_variable_set(:@@state, common_state)
+    end
+
+    after do
+      Jobs.send(:remove_const, :TestJob1)
```

Comment:
I wonder if extending `enqueue` to accept a `opts[:klass]` would be better here. Less playing around with declaring constants and removing them feels like a cleaner approach to me.

Reply:
I like this idea - much cleaner. I allowed passing a class in the `job` parameter rather than adding a new option (since the `job` parameter is required, and I didn't want to change the method signature). I updated the specs to use anonymous class definitions